mesmerize_core/caiman_extensions/common.py

```
@@ -129,37 +133,87 @@ def add_item(self, algo: str, item_name: str, input_movie_path: Union[str, pd.Se
         self._df.loc[self._df.index.size] = s
 
         # Save DataFrame to disk
-        self._df.to_pickle(self._df.paths.get_batch_path())
+        self._save_to_disk_unsafe()
 
-    def save_to_disk(self, max_index_diff: int = 0):
+    @_verify_and_lock_batch_file
+    @_index_parser
+    def update_item(self, index: int, updates: Union[dict, pd.Series]):
         """
-        Saves DataFrame to disk, copies to a backup before overwriting existing file.
+        Update the item with the given uuid with the data in updated_row and write to disk.
+
+        Parameters
+        ----------
+        index: int, str or UUID
+            The index of the batch item to update as a numerical ``int`` index, ``str`` representing
+            a UUID, or a UUID object.
+
+        updates: dict or Series
+            Data to change in the selected row. Each key in updates (or index entry if it is a Series)
+            should match a column name. If not, that data will not be saved.
```

> If not, that data will not be saved.

should probably throw an error then?

---

You're right, I was being lazy and hoped there was a way in pandas to update "strictly," but didn't find one so went with this. But I'll change it.